### ISSUE TITLE
🐛 Biome v2.2.2 crashes with index out of bounds error in module resolver

### ISSUE THREAD

**Issue by raodevendrasingh** [biomejs/biome#7355]
### Environment information

```block
CLI:
  Version:                      2.2.2
  Color support:                true

Platform:
  CPU Architecture:             x86_64
  OS:                           linux

Environment:
  BIOME_LOG_PATH:               unset
  BIOME_LOG_PREFIX_NAME:        unset
  BIOME_CONFIG_PATH:            unset
  BIOME_THREADS:                unset
  NO_COLOR:                     unset
  TERM:                         xterm-256color
  JS_RUNTIME_VERSION:           v24.7.0
  JS_RUNTIME_NAME:              node
  NODE_PACKAGE_MANAGER:         npm/11.5.2

Biome Configuration:
  Status:                       Loaded successfully
  Path:                         biome.jsonc
  Formatter enabled:            true
  Linter enabled:               true
  Assist enabled:               true
  VCS enabled:                  true

Workspace:
  Open Documents:               0
```

### What happened?

 ran `bun dev`

```
bun dev
$ NODE_OPTIONS='--max-old-space-size=4096' bun run format && 
NODE_OPTIONS='--max-old-space-size=4096' react-router dev
$ bun x ultracite format
Biome encountered an unexpected error

This is a bug in Biome, not an error in your code...(bug report rules)

Source Location: crates/biome_module_graph/src/js_module_info/module_resolver.rs:439:33
Thread Name: biome::workspace_worker_5
Message: index out of bounds: the len is 62 but the index is 1208

app/components/loaders/stateful-button.tsx internalError/panic  INTERNAL  ━━━━━━━━━━

  ✖ processing panicked: index out of bounds: the len is 62 but the index is 1208
```

[stateful-button.tsx](https://biomejs.dev/playground/?code=aQBtAHAAbwByAHQAIAB7ACAAQQByAHIAbwB3AFIAaQBnAGgAdAAsACAAQwBpAHIAYwBsAGUAQwBoAGUAYwBrACwAIABDAGkAcgBjAGwAZQBYACAAfQAgAGYAcgBvAG0AIAAiAGwAdQBjAGkAZABlAC0AcgBlAGEAYwB0ACIAOwAKAGkAbQBwAG8AcgB0ACAAdAB5AHAAZQAgAHsAIABDAG8AbQBwAG8AbgBlAG4AdABQAHIAbwBwAHMAVwBpAHQAaABvAHUAdABSAGUAZgAsACAAUgBlAGEAYwB0AE4AbwBkAGUAIAB9ACAAZgByAG8AbQAgACIAcgBlAGEAYwB0ACIAOwAKAGkAbQBwAG8AcgB0ACAAewAgAFIAaQBuAGcATABvAGEAZABlAHIAIAB9ACAAZgByAG8AbQAgACIAQAAvAGMAbwBtAHAAbwBuAGUAbgB0AHMALwBsAG8AYQBkAGUAcgBzAC8AcgBpAG4AZwAtAGwAbwBhAGQAZQByACIAOwAKAGkAbQBwAG8AcgB0ACAAewAgAEIAdQB0AHQAbwBuACAAfQAgAGYAcgBvAG0AIAAiAEAALwBjAG8AbQBwAG8AbgBlAG4AdABzAC8AdQBpAC8AYgB1AHQAdABvAG4AIgA7AAoAaQBtAHAAbwByAHQAIAB7ACAAYwBuACAAfQAgAGYAcgBvAG0AIAAiAEAALwBsAGkAYgAvAHUAdABpAGwAcwAiADsACgAKAGkAbgB0AGUAcgBmAGEAYwBlACAAUwB0AGEAdABlAGYAdQBsAEIAdQB0AHQAbwBuAFAAcgBvAHAAcwAgAGUAeAB0AGUAbgBkAHMAIABDAG8AbQBwAG8AbgBlAG4AdABQAHIAbwBwAHMAVwBpAHQAaABvAHUAdABSAGUAZgA8AHQAeQBwAGUAbwBmACAAQgB1AHQAdABvAG4APgAgAHsACgAJAGMAbABhAHMAcwBOAGEAbQBlAD8AOgAgAHMAdAByAGkAbgBnADsACgAJAGQAaQBzAGEAYgBsAGUAZAA%2FADoAIABiAG8AbwBsAGUAYQBuADsACgAJAHAAZQBuAGQAaQBuAGcAOgAgAGIAbwBvAGwAZQBhAG4AOwAKAAkAcwB0AGEAdAB1AHMAPwA6ACAAIgBpAGQAbABlACIAIAB8ACAAIgBzAHUAYwBjAGUAcwBzACIAIAB8ACAAIgBlAHIAcgBvAHIAIgA7AAoACQBwAGUAbgBkAGkAbgBnAFQAZQB4AHQAPwA6ACAAcwB0AHIAaQBuAGcAOwAKAAkAcwB1AGMAYwBlAHMAcwBUAGUAeAB0AD8AOgAgAHMAdAByAGkAbgBnADsACgAJAGUAcgByAG8AcgBUAGUAeAB0AD8AOgAgAHMAdAByAGkAbgBnADsACgAJAHMAdQBjAGMAZQBzAHMASQBjAG8AbgA%2FADoAIABSAGUAYQBjAHQATgBvAGQAZQA7AAoACQBlAHIAcgBvAHIASQBjAG8AbgA%2FADoAIABSAGUAYQBjAHQATgBvAGQAZQA7AAoACQBjAGgAaQBsAGQAcgBlAG4AOgAgAFIAZQBhAGMAdABOAG8AZABlADsACgAJAG8AbgBDAGwAaQBjAGsAPwA6ACAAKAApACAAPQA%2BACAAdgBvAGkAZAA7AAoACQBmAG8AcgBtAD8AOgAgAHMAdAByAGkAbgBnADsACgAJAHQAeQBwAGUAPwA6ACAAIgBzAHUAYgBtAGkAdAAiACAAfAAgACIAYgB1AHQAdABvAG4AIgAgAHwAIAAiAHIAZQBzAGUAdAAiADsACgB9AAoACgBlAHgAcABvAHIAdAAgAGYAdQBuAGMAdABpAG8AbgAgAFMAdABhAHQAZQBmAHUAbABCAHUAdAB0AG8AbgAoAHsACgAJAGMAbABhAHMAcwBOAGEAbQBlACwACgAJAGQAaQBzAGEAYgBsAGUAZAAsAAoACQBwAGUAbgBkAGkAbgBnACwACgAJAHMAdABhAHQAdQBzACAAPQAgACIAaQBkAGwAZQAiACwACgAJAHAAZQBuAGQAaQBuAGcAVABlAHgAdAAsAAoACQBzAHUAYwBjAGUAcwBzAFQAZQB4AHQALAAKAAkAcwB1AGMAYwBlAHMAcwBJAGMAbwBuACwACgAJAGUAcgByAG8AcgBUAGUAeAB0ACwACgAJAGUAcgByAG8AcgBJAGMAbwBuACwACgAJAGMAaABpAGwAZAByAGUAbgAsAAoACQBvAG4AQwBsAGkAYwBrACwACgAJAGYAbwByAG0ALAAKAAkAdAB5AHAAZQAgAD0AIAAiAHMAdQBiAG0AaQB0ACIALAAKAAkALgAuAC4AcAByAG8AcABzAAoAfQA6ACAAUwB0AGEAdABlAGYAdQBsAEIAdQB0AHQAbwBuAFAAcgBvAHAAcwApACAAewAKAAkAYwBvAG4AcwB0ACAAaQBzAEQAaQBzAGEAYgBsAGUAZAAgAD0AIABkAGkAcwBhAGIAbABlAGQAIAB8AHwAIABwAGUAbgBkAGkAbgBnADsACgAJAGMAbwBuAHMAdAAgAGkAcwBTAHUAYwBjAGUAcwBzACAAPQAgAHMAdABhAHQAdQBzACAAPQA9AD0AIAAiAHMAdQBjAGMAZQBzAHMAIgAgACYAJgAgACEAcABlAG4AZABpAG4AZwA7AAoACQBjAG8AbgBzAHQAIABpAHMARQByAHIAbwByACAAPQAgAHMAdABhAHQAdQBzACAAPQA9AD0AIAAiAGUAcgByAG8AcgAiACAAJgAmACAAIQBwAGUAbgBkAGkAbgBnADsACgAJAGMAbwBuAHMAdAAgAGkAcwBJAGQAbABlACAAPQAgACEAKABwAGUAbgBkAGkAbgBnACAAfAB8ACAAaQBzAFMAdQBjAGMAZQBzAHMAIAB8AHwAIABpAHMARQByAHIAbwByACkAOwAKAAoACQBjAG8AbgBzAHQAIABnAGUAdABCAHUAdAB0AG8AbgBDAG8AbgB0AGUAbgB0ACAAPQAgACgAKQAgAD0APgAgAHsACgAJAAkAaQBmACAAKABwAGUAbgBkAGkAbgBnACkAIAB7AAoACQAJAAkAcgBlAHQAdQByAG4AIAAoAAoACQAJAAkACQA8AHMAcABhAG4AIABjAGwAYQBzAHMATgBhAG0AZQA9ACIAaQBuAGwAaQBuAGUALQBmAGwAZQB4ACAAaQB0AGUAbQBzAC0AYwBlAG4AdABlAHIAIABnAGEAcAAtADIAIgA%2BAAoACQAJAAkACQAJAHsAcABlAG4AZABpAG4AZwBUAGUAeAB0ACAAfAB8ACAAIgBMAG8AYQBkAGkAbgBnACIAfQAKAAkACQAJAAkACQA8AFIAaQBuAGcATABvAGEAZABlAHIAIAAvAD4ACgAJAAkACQAJADwALwBzAHAAYQBuAD4ACgAJAAkACQApADsACgAJAAkAfQAKAAoACQAJAGkAZgAgACgAaQBzAFMAdQBjAGMAZQBzAHMAKQAgAHsACgAJAAkACQByAGUAdAB1AHIAbgAgACgACgAJAAkACQAJADwAcwBwAGEAbgAgAGMAbABhAHMAcwBOAGEAbQBlAD0AIgBpAG4AbABpAG4AZQAtAGYAbABlAHgAIABpAHQAZQBtAHMALQBjAGUAbgB0AGUAcgAgAGcAYQBwAC0AMgAiAD4ACgAJAAkACQAJAAkAewBzAHUAYwBjAGUAcwBzAFQAZQB4AHQAIAB8AHwAIABjAGgAaQBsAGQAcgBlAG4AfQAKAAkACQAJAAkACQB7AHMAdQBjAGMAZQBzAHMASQBjAG8AbgAgAHwAfAAgADwAQwBpAHIAYwBsAGUAQwBoAGUAYwBrACAAYwBsAGEAcwBzAE4AYQBtAGUAPQAiAGgALQA0ACAAdwAtADQAIgAgAC8APgB9AAoACQAJAAkACQA8AC8AcwBwAGEAbgA%2BAAoACQAJAAkAKQA7AAoACQAJAH0ACgAKAAkACQBpAGYAIAAoAGkAcwBFAHIAcgBvAHIAKQAgAHsACgAJAAkACQByAGUAdAB1AHIAbgAgACgACgAJAAkACQAJADwAcwBwAGEAbgAgAGMAbABhAHMAcwBOAGEAbQBlAD0AIgBpAG4AbABpAG4AZQAtAGYAbABlAHgAIABpAHQAZQBtAHMALQBjAGUAbgB0AGUAcgAgAGcAYQBwAC0AMgAiAD4ACgAJAAkACQAJAAkAewBlAHIAcgBvAHIAVABlAHgAdAAgAHwAfAAgACIARQByAHIAbwByACIAfQAKAAkACQAJAAkACQB7AGUAcgByAG8AcgBJAGMAbwBuACAAfAB8ACAAPABDAGkAcgBjAGwAZQBYACAAYwBsAGEAcwBzAE4AYQBtAGUAPQAiAGgALQA0ACAAdwAtADQAIgAgAC8APgB9AAoACQAJAAkACQA8AC8AcwBwAGEAbgA%2BAAoACQAJAAkAKQA7AAoACQAJAH0ACgAKAAkACQBpAGYAIAAoAGkAcwBJAGQAbABlACkAIAB7AAoACQAJAAkAcgBlAHQAdQByAG4AIAAoAAoACQAJAAkACQA8AHMAcABhAG4AIABjAGwAYQBzAHMATgBhAG0AZQA9ACIAZgBsAGUAeAAgAGkAdABlAG0AcwAtAGMAZQBuAHQAZQByACAAZwBhAHAALQAyACIAPgAKAAkACQAJAAkACQB7AGMAaABpAGwAZAByAGUAbgB9AAoACQAJAAkACQAJADwAQQByAHIAbwB3AFIAaQBnAGgAdAAgAGMAbABhAHMAcwBOAGEAbQBlAD0AIgBoAC0ANAAgAHcALQA0ACAAbwBwAGEAYwBpAHQAeQAtADcAMAAgAHQAcgBhAG4AcwBpAHQAaQBvAG4ALQBvAHAAYQBjAGkAdAB5ACAAZAB1AHIAYQB0AGkAbwBuAC0AMgAwADAAIABnAHIAbwB1AHAALQBoAG8AdgBlAHIAOgBvAHAAYQBjAGkAdAB5AC0AMQAwADAAIgAgAC8APgAKAAkACQAJAAkAPAAvAHMAcABhAG4APgAKAAkACQAJACkAOwAKAAkACQB9AAoACQB9ADsACgAKAAkAcgBlAHQAdQByAG4AIAAoAAoACQAJADwAQgB1AHQAdABvAG4ACgAJAAkACQBjAGwAYQBzAHMATgBhAG0AZQA9AHsAYwBuACgACgAJAAkACQAJACIAZwByAG8AdQBwACAAdwAtAGYAdQBsAGwAIAByAG8AdQBuAGQAZQBkAC0AbABnACAAZABpAHMAYQBiAGwAZQBkADoAYwB1AHIAcwBvAHIALQBuAG8AdAAtAGEAbABsAG8AdwBlAGQAIABkAGkAcwBhAGIAbABlAGQAOgBvAHAAYQBjAGkAdAB5AC0ANQAwACIALAAKAAkACQAJAAkAaQBzAFMAdQBjAGMAZQBzAHMAIAAmACYAIAAiAGIAZwAtAGcAcgBlAGUAbgAtADYAMAAwACAAdABlAHgAdAAtAHcAaABpAHQAZQAgAGgAbwB2AGUAcgA6AGIAZwAtAGcAcgBlAGUAbgAtADcAMAAwACIALAAKAAkACQAJAAkAaQBzAEUAcgByAG8AcgAgACYAJgAgACIAYgBnAC0AcgBvAHMAZQAtADYAMAAwACAAdABlAHgAdAAtAHcAaABpAHQAZQAgAGgAbwB2AGUAcgA6AGIAZwAtAHIAbwBzAGUALQA3ADAAMAAiACwACgAJAAkACQAJAGMAbABhAHMAcwBOAGEAbQBlAAoACQAJAAkAKQB9AAoACQAJAAkAZABpAHMAYQBiAGwAZQBkAD0AewBpAHMARABpAHMAYQBiAGwAZQBkAH0ACgAJAAkACQBmAG8AcgBtAD0AewBmAG8AcgBtAH0ACgAJAAkACQBvAG4AQwBsAGkAYwBrAD0AewBvAG4AQwBsAGkAYwBrAH0ACgAJAAkACQB0AHkAcABlAD0AewB0AHkAcABlAH0ACgAJAAkACQB7AC4ALgAuAHAAcgBvAHAAcwB9AAoACQAJAD4ACgAJAAkACQB7AGcAZQB0AEIAdQB0AHQAbwBuAEMAbwBuAHQAZQBuAHQAKAApAH0ACgAJAAkAPAAvAEIAdQB0AHQAbwBuAD4ACgAJACkAOwAKAH0ACgA%3D)

> The component itself appears syntactically correct

#### In App notifications in Cursor IDE:
* Client biome: connection to server is erroring. Cannot call write after a stream was destroyed.
* Client biome: connection to server is erroring. write EPIPE Shutting down server.

### Expected result

It shouldn't throw any error.

### Code of Conduct

- [x] I agree to follow Biome's Code of Conduct
